pkg/resources/rbac.authorization.k8s.io/v1/clusterrole/validator_test.go

```diff
@@ -0,0 +1,172 @@
+package clusterrole
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+
+	"github.com/rancher/webhook/pkg/admission"
+	"github.com/stretchr/testify/require"
+	admissionv1 "k8s.io/api/admission/v1"
+	v1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var (
+	gvk = metav1.GroupVersionKind{Group: "rbac.authorization.k8s.io", Version: "v1", Kind: "Role"}
+	gvr = metav1.GroupVersionResource{Group: "rbac.authorization.k8s.io", Version: "v1", Resource: "roles"}
+)
+
+func TestAdmit(t *testing.T) {
+	t.Parallel()
+
+	defaultRole := &v1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "default",
+		},
+	}
+	emptyRole := &v1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: map[string]string{},
+		},
+	}
+	roleWithOwnerLabel := &v1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "default",
+			Labels: map[string]string{
+				grOwnerLabel: "gr-owner",
+			},
+		},
+	}
+	roleWithNewOwnerLabel := &v1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "default",
+			Labels: map[string]string{
+				grOwnerLabel: "new-owner",
+			},
+		},
+	}
+
+	type args struct {
+		oldRole *v1.ClusterRole
+		newRole *v1.ClusterRole
+	}
+	type test struct {
+		name    string
+		args    args
+		allowed bool
+		wantErr bool
+	}
+	tests := []test{
```

Comment:
Would be good to add a test for the error case (where it can't pull the old/new rule from the request). Example [here](https://github.com/rancher/webhook/blob/8e3d96998fe1d95e71196c06d8c6dca26e284282/pkg/resources/core/v1/secret/validator_test.go#L212)

Reply:
added